### ISSUE TITLE
shufflenet v2+ssd  performance loss too much

### ISSUE THREAD

**Issue by DL-85** [BUG1989/caffe-int8-convert-tools#10]
Thanks for the great tools!
When I use this tool  to generate table from shufflenet v2+ssd prototxt and caffemodel, and use the caffe2ncnn tool to get int8 model, the performance loss too much, but mobilenet v1+ssd is fine with this tool. Why is it, any idea?

**Comment by DL-85** [comment]
Do I have to merge the BatchNorm layer and Scale layer to Convolution layer (modified the bias_term: to true and convert the caffemodel) before use this tool?

**Comment by BUG1989** [comment]
This effect will be better

**Comment by leeburt** [comment]
@DL-85 hi，i follow the step to convert my mobilenet-ssd to int8 model,but the cost time is more and the performance is bad. Do you know the reason？It is very grateful if you give me any suggest.

**Comment by BUG1989** [comment]
For this model(mobilenet_v1_ssd 300),I find that if you close the conv1 layer Int8 forward,and using Float32 forward inplace it,the mAP maybe increase.Please help me confirm it.

**Comment by titikid** [comment]
@BUG1989 How do you know it?
Anw, i forgot to modify confident_threshold param in file mobilenet_ssd_voc_ncnn.param. Here are updated result for mobilenet_ssd_300:
- fp32 model: mAP: 72.33%
- int8 model: mAP=60.15%

**Comment by BUG1989** [comment]
@titikid We tried it out layer by layer ^_^

**Comment by BUG1989** [comment]
Recently i have been trying a new implement : kld + fine-tuning.It can reduce the precision loss(mAP loss < %).But i do the work in my spare time, so I don't know when I can finish it (

**Comment by titikid** [comment]
@BUG1989 yes.kld+fine-tuning should be better.
Sorry. This is my first time to use ncnn framework. How can i "close the conv1 layer Int8 forward,and using Float32 forward inplace it" as your recommendation?

**Comment by BUG1989** [comment]
@titikid [mixed-precision-inference](https://github.com/Tencent/ncnn/wiki/quantized-int8-inference#mixed-precision-inference)

**Comment by titikid** [comment]
@BUG1989 is there some mistake here: "before converting your model files, delete the layer weight scale line in table file, and that layer will do the float32 inference" ???. i think table file is output of converting process.

**Comment by titikid** [comment]
@BUG1989 i add 7=1 in conv1 in param file and the accuracy is significantly increase (mAP=66.35%). Cheer!

**Comment by BUG1989** [comment]
@titikid Thank you very much for helping me with this experiment

**Comment by BUG1989** [comment]
@DL-85 @titikid I test a new case,with output channel split,[k, k, cin, cout] cut into "cout" buckets.quantize all layer in mobilenet_v1_ssd,the mAP from  60.38 to 68.85(the mAP of FP32 is 70.49)

**Comment by titikid** [comment]
@BUG1989 Do you have any document for new quantization method (base on number of group)? btw, i couldn't see any MobileNet_v1_dev.table in repository

**Comment by BUG1989** [comment]
@titikid Hi,I have update those files,base on number of output channel number.

- using [caffe-int8-convert-tools-dev-weight.py](https://github.com/BUG1989/caffe-int8-convert-tools/blob/master/caffe-int8-convert-tool-dev-weight.py) to generate the calibration.table,maybe you can use those [tables](https://github.com/BUG1989/caffe-int8-convert-tools/tree/master/temp) ;
- using this [ncnn](https://github.com/BUG1989/ncnn/tree/ncnn-int8-perchannel) to adapt it(caffe2ncnn,examples,and so on).

**Comment by titikid** [comment]
@BUG1989 i will try it. Thanks for your work!

**Comment by shanshan0412** [comment]
> @DL-85 @titikid I test a new case,with output channel split,[k, k, cin, cout] cut into "cout" buckets.quantize all layer in mobilenet_v1_ssd,the mAP from 60.38 to 68.85(the mAP of FP32 is 70.4)

您好，能具体描述下这个方法的操作步骤吗？谢谢！

**Comment by andeyeluguo** [comment]
how did you get the quantization performance? @titikid

**Comment by andeyeluguo** [comment]
what do you mean "add 7=1 in conv1 in param file"
